### PR TITLE
Typescript

### PR DESCRIPTION
**Pull request by CSMastermind**

### CODE REVIEW

File: .idea/typescript-compiler.xml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
```
Comment (hackerrdave) [inline .idea/typescript-compiler.xml]: why are we adding this file?

File: index.js [hunk 2]
```diff
@@ -657,7 +662,9 @@ RemoteObjectTemplate.serializeAndGarbageCollect = function serializeAndGarbageCo
 
     function serialize(obj) {
         try {
-            return JSON.stringify(obj, function y(_key, value) {
+            return JSON.stringify(obj, function y(key, value) {
+                if (key == '__objectTemplate__')
+                    return null;
```
Comment (hackerrdave) [inline index.js]: missing `{}`

File: test/supertype/Role.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
+"use strict";
+var __extends = (this && this.__extends) || (function () {
```
Comment (hackerrdave) [inline test/supertype/Role.js]: what is this file for?

File: test/supertype/Role.js.map [hunk 4]
```diff
@@ -0,0 +1 @@
+{"version":3,"file":"Role.js","sourceRoot":"","sources":["Role.ts"],"names":[],"mappings":";;;;;;;;;;;;;;;;;;;;;AAAA,qCAAgE;AAChE,uCAAoC;AACpC,qCAAkC;AAGlC,IAAa,IAAI;IAAS,wBAAS;IAE/B,cAAa,QAAQ,EAAE,OAAO,EAAE,YAAY;QAA5C,YACI,iBAAO,SAKV;QAGD,kBAAY,GAAW,SAAS,CAAC;QAP7B,KAAI,CAAC,QAAQ,GAAG,QAAQ,CAAC;QACzB,KAAI,CAAC,OAAO,GAAG,OAAO,CAAC;QACvB,EAAE,CAAC,CAAC,YAAY,CAAC;YACb,KAAI,CAAC,YAAY,GAAG,YAAY,CAAC;;IACzC,CAAC;IAAA,CAAC;IAUN,WAAC;AAAD,CAAC,AAlBD,CAA0B,iBAAS,GAkBlC;AAPG;IADC,gBAAQ,EAAE;;0CACsB;AAGjC;IADC,gBAAQ,CAAC,EAAC,OAAO,EAAE,cAAO,MAAM,CAAC,mBAAQ,CAAA,CAAA,CAAC,EAAC,CAAC;8BACnC,mBAAQ;sCAAC;AAGnB;IADC,gBAAQ,CAAC,EAAC,OAAO,EAAE,cAAO,MAAM,CAAC,iBAAO,CAAA,CAAA,CAAC,EAAC,CAAC;8BACnC,iBAAO;qCAAC;AAjBR,IAAI;IADhB,sBAAc;;GACF,IAAI,CAkBhB;AAlBY,oBAAI"}
```
Comment (hackerrdave) [inline test/supertype/Role.js.map]: why checking in sourcemap?

**Review by hackerrdave — COMMENTED** [review]
Should we be checking in the compiled files?

File: .idea/typescript-compiler.xml (hunk 1 above)
Comment (CSMastermind) [reply .idea/typescript-compiler.xml]: Because the .gitignore didn't catch it.  Removing now.

File: index.js (hunk 2 above)
Comment (CSMastermind) [reply index.js]: Fixed

File: test/supertype/Role.js.map (hunk 4 above)
Comment (CSMastermind) [reply test/supertype/Role.js.map]: We should not be.  Fixing.

**Pull request merged**